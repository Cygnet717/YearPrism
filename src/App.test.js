import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import UserProvider from './Context/user-provider'

describe(`App rendering`, () => {
  it(`renders without crashing`, () => {
    const div = document.createElement('div');
    ReactDOM.render(<UserProvider />, div);
    ReactDOM.unmountComponentAtNode(div);
});
it('renders the UI as expected', () => {
  const tree= renderer
  .create(<UserProvider name="App" />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
it(`renders DeleteAccount without crashing`, () => {
  const div = document.createElement('div');
  ReactDOM.render(<UserProvider name="DeleteAccount"/>, div);
  ReactDOM.unmountComponentAtNode(div);
});
it('renders the footer as expected', () => {
  const tree= renderer
  .create(<UserProvider name="Footer" />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
it('renders the header as expected', () => {
  const tree= renderer
  .create(<UserProvider name="Header" />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
it('renders the HomePage as expected', () => {
  const tree= renderer
  .create(<UserProvider name="HomePage" />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
it('renders the LandingPage as expected', () => {
  const tree= renderer
  .create(<UserProvider name="LandingPage" />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
it('renders the LoginPage as expected', () => {
  const tree= renderer
  .create(<UserProvider name="LoginPage" />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
it('renders the RegisterPage as expected', () => {
  const tree= renderer
  .create(<UserProvider name="RegisterPage" />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
it('renders the SearchResults as expected', () => {
  const tree= renderer
  .create(<UserProvider name="SearchResults" />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
it('renders the YearViewPage as expected', () => {
  const tree= renderer
  .create(<UserProvider name="YearViewPage" />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
})