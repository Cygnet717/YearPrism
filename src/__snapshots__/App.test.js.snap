// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering renders the HomePage as expected 1`] = `null`;

exports[`App rendering renders the LandingPage as expected 1`] = `null`;

exports[`App rendering renders the LoginPage as expected 1`] = `null`;

exports[`App rendering renders the RegisterPage as expected 1`] = `null`;

exports[`App rendering renders the SearchResults as expected 1`] = `null`;

exports[`App rendering renders the UI as expected 1`] = `null`;

exports[`App rendering renders the YearViewPage as expected 1`] = `null`;

exports[`App rendering renders the footer as expected 1`] = `null`;

exports[`App rendering renders the header as expected 1`] = `null`;
